overlord/snapshotstate/backend/backend.go

```
@@ -544,13 +568,72 @@ func (t *importTransaction) Cancel() error {
 
 // Commit will commit a given transaction
 func (t *importTransaction) Commit() error {
-	if err := os.Remove(importInProgressFilepath(t.id)); err != nil {
+	t.committed = true
+	return t.unlock()
+}
+
+func (t *importTransaction) lock() error {
```

Maybe throw an internal error if t.flock is already set?